healme.lic

```diff
@@ -2,198 +2,320 @@
   Documentation: https://elanthipedia.play.net/Lich_script_repository#healme
 =end
 
-# ***BEGIN USER CONFIG***
-
-UserVars.healme_debug ||= false # turn on echos in the script
-
-# ***END USER CONFIG***
-
-custom_require.call(%w[common common-arcana common-healing events])
+custom_require.call(%w[common common-arcana common-healing events spellmonitor])
 
 class HealMe
   include DRC
   include DRCA
   include DRCH
 
   def initialize
-    @deadly_wounds = %w[SKIN HEAD NECK CHEST ABDOMEN BACK]
     @settings = get_settings
-
     arg_definitions = [
       [
-        { name: 'bleeders', regex: /^[\w\s,]+$/i, variable: true, optional: true, description: 'comma separated list of bleeders to keep. "right arm, chest, back"' }
+        { name: 'debug', regex: /debug/i, optional: true, description: 'Enable debug output' },
+        { name: 'keep', regex: /^[\w\s,]+$/i, variable: true, optional: true, description: 'Comma separated list of wounds to keep. Example: "right arm, chest, back"' }
       ]
     ]
-
     args = parse_args(arg_definitions)
-    @using_waggle = false
-
-    heal_self(args.bleeders)
+    $debug_mode_hm = UserVars.healme_debug || args.debug
+    @spells = load_healing_waggle_set
+    @body_parts_not_to_heal = args.keep.split(',').map(&:strip) || []
+    ['internal', 'external'].each { |x| ['wounds', 'scars'].each { |y| Flags.add("hm-#{x}-#{y}-healed", "The #{x} #{y} on your (.+) appear completely healed") } }
+    heal_self
   end
 
-  def severity(wounds)
-    wounds.map { |wound| wound['level'] }.inject(&:+)
+  # The original version of the 'healme' script used
+  # the 'empath_healing:' setting to configure the spells to cast.
+  # Later, the script added support for waggle sets.
+  # This method checks for how the player has specified their
+  # healing spells, using the waggle set by default if defined.
+  # Returns the spell data in a waggle format, regardless the configuration used.
+  def load_healing_waggle_set
```

Comment:
I like this. I basically decided that I wasn't going to support empath_healing in my version and just dropped it, but this is a nicer backwards compat option. Since I think combat-trainer's empath healing needs to be reworked eventually, maybe we can eventually deprecate this all together.